test/functional/normalization/json_format_test.go

```diff
@@ -0,0 +1,52 @@
+package normalization
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/test/functional"
+	"github.com/openshift/cluster-logging-operator/test/helpers/types"
+	"github.com/openshift/cluster-logging-operator/test/matchers"
+	"strings"
+)
+
+var _ = Describe("[Functional][Normalization][Json] Parse json format log", func() {
+
+	var (
+		framework *functional.FluentdFunctionalFramework
+		// json message
+		json = "{\"name\":\"fred\",\"home\":\"bedrock\"}"
+		// Template expected as output Log
+		outputLogTemplate = functional.NewApplicationLogTempate()
+	)
+
+	BeforeEach(func() {
+
+		framework = functional.NewFluentdFunctionalFramework()
+		functional.NewClusterLogForwarderBuilder(framework.Forwarder).
+			FromInput(logging.InputNameApplication).
+			ToFluentForwardOutput()
+		Expect(framework.Deploy()).To(BeNil())
+		Expect(framework.WritesJsonApplicationLogs(json, 1)).To(BeNil())
+	})
+	AfterEach(func() {
+		framework.Cleanup()
+	})
+
+	Context("when parsing json logs as un-structured", func() {
+		It("should be logged as a message", func() {
+			outputLogTemplate.Message = strings.Replace(json, "\"", "", -1)
```

Comment:
What exactly are we verifying here.  Initially as-is, the message field is a string representation of JSON.  What do we believe the message field looks like here?

Reply:
@jcantrill you will be suprised .... what I saw is that we are getting an empty message ... and that the log message is added as Json key+value to the output ... not sure if anyone was planning on that ... anyway ... we will add more here over time ... let me change the text to call this WIP

Reply:
@jcantrill I updated the PR ... now if you look on option #2 you will see here: https://github.com/openshift/cluster-logging-operator/pull/932/files#diff-6bcd37f08b54c03e0e0a44f0cce637f8d3b13a3c15661aef973a76556d57e7f5R413 

The "leading" plan for the new behaiviour after API will be implmented for this feature ... this is the expected fluentd conf file (for now) This test compares current behviour and future behviour, after we will have support for json logs

Hope this make sense ???

Reply:
@alanconway ^^^ FYI